### PR TITLE
Adding tab navigation to Org and Profile screen

### PR DESCRIPTION
**Pull request by AllStackDev1**
Closes: #1696
- Added Org and Profile Tab nav.
- Made org repo filter sticky.
- wasn't able to make the filter sticky in the profile screen as the layout is complex.

Simulator Screen Shot - iPhone 14 - 2023-03-30 at 05 20 26

Screenshot 2023-03-30 at 5 20 51 AM

### CODE REVIEW

File: expo-zustand-styled-components/src/screens/Profile/Profile.tsx [hunk 1]
```diff
@@ -1,60 +1,75 @@
...
 
-const Profile = ({ route }: AppStackScreenProps<'Profile'>) => {
-  const { width } = useWindowDimensions();
+import { AppStackScreenProps } from '../../../types';
+import { Platform } from 'react-native';
 
+const Profile = ({ route, navigation }: AppStackScreenProps<'Profile'>) => {
+  const [leftPadding, setLeftPadding] = useState(0);
+  const { width, height } = useWindowDimensions();
   const { user, error, isLoading } = useAuthStore();
```
Comment (hdJerry) [inline expo-zustand-styled-components/src/screens/Profile/Profile.tsx:23]:
```suggestion
 interface Elm extends Element {
    style: {
      zIndex: string;
    }
  }
```

File: expo-zustand-styled-components/src/screens/Profile/Profile.tsx [hunk 2]
```diff
@@ -1,60 +1,75 @@
...
     getUserProfile({ username: route.params.username });
   }, [route.params.username]);
 
+  useLayoutEffect(() => {
+    navigation.setOptions({ title: `Profile . ${route.params.username}` });
+  }, [navigation, route.params]);
+
+  useEffect(() => {
+    if (Platform.OS === 'web') {
+      document.querySelectorAll('[class*=css-view-]').forEach((e) => {
```
Comment (hdJerry) [inline expo-zustand-styled-components/src/screens/Profile/Profile.tsx:34]:
```suggestion
      document.querySelectorAll('[class*=css-view-]').forEach((e: Elm) => {
```

File: expo-zustand-styled-components/src/screens/Profile/Profile.tsx [hunk 3]
```diff
@@ -1,60 +1,75 @@
...
+        // we want to skip elements that have a z-index class
+        if (!/r-zIndex-/.test(e.className)) {
+          // check if element has a direct child of type 'img'
+          if (!e.querySelectorAll(':scope > img')?.length) {
+            // if yes, unset the z-index. This is to prevent the image from
+            // being hidden behind the header when the user scrolls up
+            // @ts-ignore
```
[the remark this answers is not in the document]
Comment (hdJerry) [reply expo-zustand-styled-components/src/screens/Profile/Profile.tsx:41]: you can now remove this.

File: expo-zustand-styled-components/src/components/TabNavigation/TabNavigation.tsx [hunk 4]
```diff
@@ -38,7 +43,7 @@ const TabNavigation = ({ tabs, activeTab, onChange }: TabNavigationProps) => {
             key={index}
             activeOpacity={0.5}
             isActive={activeTab === title}
-            onPress={() => onChange(title)}>
+            onPress={() => onChange ? onChange(title) : {}}>
```
Comment (hdJerry) [inline expo-zustand-styled-components/src/components/TabNavigation/TabNavigation.tsx:46]:
```suggestion
            onPress={() => onChange ? onChange(title) : null}>
```

**Review by hdJerry — CHANGES_REQUESTED** [review]
Screenshot 2023-03-30 at 11 25 58

This doesn't seem to scroll on mobile.

**Review by hdJerry — CHANGES_REQUESTED** [review]
I thought the goal was to make the filter stuff sticky. This isn't sticky

Screenshot 2023-03-30 at 11 27 37

File: expo-zustand-styled-components/src/components/Profile/UserCard/UserCard.styles.ts [hunk 5]
```diff
@@ -6,25 +6,26 @@ type ScreenWidth = {
...
-  margin-top: ${({ screenWidth }) => (screenWidth >= breakpoints.tablet ? -20 : 0)}px;
+  z-index: unset;
   width: ${({ screenWidth }) => (screenWidth >= breakpoints.tablet ? 33.3 : 100)}%;
+  margin-top: ${({ screenWidth }) => (screenWidth >= breakpoints.tablet ? -80 : 0)}px;
+  padding-right: ${({ screenWidth }) => (screenWidth >= breakpoints.tablet ? 2.5 : 0)}%;
 `;
 
 export const UserDetails = styled.View<ScreenWidth>`
   gap: 8px;
+  z-index: unset;
```
Comment (hdJerry) [inline expo-zustand-styled-components/src/components/Profile/UserCard/UserCard.styles.ts:17]: Expected style "zIndex: unset" to be unitless 
That's the error

**Review by hdJerry — CHANGES_REQUESTED** [review]

File: expo-zustand-styled-components/src/components/Profile/UserCard/UserCard.styles.ts (hunk 5 above)
Comment (AllStackDev1) [reply expo-zustand-styled-components/src/components/Profile/UserCard/UserCard.styles.ts:17]: This is a warning from the style component that doesn't make any sense. as z-index can be a text, but then it is expecting it to be just numbered. This warning also appears in the web console. but it isn't an error.

Comment (hdJerry) [reply expo-zustand-styled-components/src/components/Profile/UserCard/UserCard.styles.ts:17]: It's failing to load on my phone

Comment (hdJerry) [reply expo-zustand-styled-components/src/components/Profile/UserCard/UserCard.styles.ts:17]: Did you test with your phone?

Comment (hdJerry) [reply expo-zustand-styled-components/src/components/Profile/UserCard/UserCard.styles.ts:17]: And? WHat did you see?

**Review by hdJerry — DISMISSED** [review]

**Review by hdJerry — DISMISSED** [review]
The profile filter is not sticky, any reason why? since that is one of the essence of this PR

File: expo-zustand-styled-components/src/components/Profile/UserCard/UserCard.styles.ts (hunk 5 above)
Comment (AllStackDev1) [reply expo-zustand-styled-components/src/components/Profile/UserCard/UserCard.styles.ts]: No issue on my end put i have been able to remove it for mobile

**Comment by AllStackDev1** [comment]
> The profile filter is not sticky, any reason why? since that is one of the essence of this PR

here is -> https://github.com/thisdot/starter.dev-github-showcases/pull/1713#issuecomment-1490182604

**Review by markshenouda — APPROVED** [review]

**Pull request merged**